src/main/java/org/dependencytrack/common/ConfigKey.java

```diff
@@ -5,14 +5,18 @@
 import java.time.Duration;
 
 public enum ConfigKey implements Config.Key {
+
     SNYK_THREAD_BATCH_SIZE("snyk.thread.batch.size", 10),
     SNYK_LIMIT_FOR_PERIOD("snyk.limit.for.period", 1500),
     SNYK_THREAD_TIMEOUT_DURATION("snyk.thread.timeout.duration", 60),
     SNYK_LIMIT_REFRESH_PERIOD("snyk.limit.refresh.period", 60),
     OSSINDEX_REQUEST_MAX_PURL("ossindex.request.max.purl", 128),
     OSSINDEX_RETRY_EXPONENTIAL_BACKOFF_MAX_ATTEMPTS("ossindex.retry.backoff.max.attempts", 10),
     OSSINDEX_RETRY_EXPONENTIAL_BACKOFF_MULTIPLIER("ossindex.retry.backoff.multiplier", 2),
-    OSSINDEX_RETRY_EXPONENTIAL_BACKOFF_MAX_DURATION("ossindex.retry.backoff.max.duration", Duration.ofMinutes(10).toMillis());
+    OSSINDEX_RETRY_EXPONENTIAL_BACKOFF_MAX_DURATION("ossindex.retry.backoff.max.duration", Duration.ofMinutes(10).toMillis()),
+    REPO_META_ANALYZER_CACHE_STAMPEDE_BLOCKER_ENABLED("repo.meta.analyzer.cacheStampedeBlocker.enabled", false),
+    REPO_META_ANALYZER_CACHE_STAMPEDE_BLOCKER_LOCK_BUCKETS("repo.meta.analyzer.cacheStampedeBlocker.lock.buckets", 1000),
```

Comment:
After profiling the memory, seems like the footprint of the striped lock would be 32B * (buckets + 1).
Something in between 1000 (32 KB) & 1000 000 (32 MB) seems reasonable in the memory front.
As for the optimal count, only test could tell 😅

Reply:
Would it be safe to assume that the bucket size should be roughly in the same ballpark as the number of threads accessing the protected resources? Say I have a worker pool of 40, I wouldn't gain much from having 1000 locks, as it is unlikely that more than >40 threads would try to access the same resource.

The [guava docs](https://guava.dev/releases/19.0/api/docs/com/google/common/util/concurrent/Striped.html) say:

> For example, if a set of tasks are CPU-bound, one could easily create a very compact Striped<Lock> of availableProcessors() * 4 stripes

Not 100% sure how this adapts to more I/O bound tasks like in our case.

Anyway, I agree that the footprint of 1000 is relatively small, and we can always revisit this at a later point in time.

Reply:
> Say I have a worker pool of 40, I wouldn't gain much from having 1000 locks, as it is unlikely that more than >40 threads would try to access the same resource. 

Yeah but this doesn't account for purl distribution, it could be different purls but worst case scenario would be that all those threads end up requesting the same inner lock from the striped lock. A bigger bucket size make it less probable.

> Not 100% sure how this adapts to more I/O bound tasks like in our case.

Same here. Let's try and see.